pkg/apiserver/profiling/router.go

```diff
@@ -303,7 +311,29 @@ func (s *Service) viewSingle(c *gin.Context) {
 		_ = c.Error(err)
 		return
 	}
-	c.Data(http.StatusOK, "image/svg+xml", content)
+
+	// set default content-type for legacy profiling content.
+	contentType := "image/svg+xml"
+
+	if task.ProfileOutputType == ProfilingOutputTypeProtobuf {
+		switch outputType {
+		case string(ViewOutputTypeGraph):
+			svgContent, err := convertProtobufToSVG(content, task)
+			if err != nil {
+				_ = c.Error(err)
+				return
+			}
+			content = svgContent
+			contentType = "image/svg+xml"
+		case string(ViewOutputTypeFlameGraph):
+			contentType = "application/protobuf"
+		default:
+			// Will not handle converting protobuf to other formats except flamegraph and graph
+			log.Warn("Failed to convert protobuf to %v", zap.String("output", outputType))
```

Comment:
In this case you can use:

```go
_ = c.Error(rest.ErrBadRequest.New("Cannot output protobuf as %s", outputType))
return
```

so that from the API perspective user can receive a standard error response with a correct BadRequest status code.